src/app/commands/cmd_remove_frame.cpp

```diff
@@ -59,6 +59,11 @@ void RemoveFrameCommand::onExecute(Context* context)
     if (site->inTimeline() &&
         !site->selectedFrames().empty()) {
       for (frame_t frame : site->selectedFrames().reversed()) {
+
+        // if we are on the first frame - don't do anything with it
+        if (frame == 0)
+          continue;
```

Comment:
If we select all frames and delete them, they should be deleted, i.e. a frame without cel or a background layer cleared with the background color is just left (it's just like creating a new sprite with the given layers empty and just one frame).

With this change the first frame is kept (which is not expected).

From #3551 the bug is that the background layer is left without a cel (which is inconsistent with the background layer which must have one cel on each frame).

Reply:
Hm, so if I understand correctly - the issue is actually not the cel being removed from the frame, but ONLY the cel being removed from the frame, not the frame itself?

So if I understand correctly - we should remove frame + cel inside of it, right?

Reply:
> So if I understand correctly - we should remove frame + cel inside of it, right? So if we remove all frames, there should be no frame at all, right?

No. The issue is that for the background layer (and only for the background layer) the cel is being deleted instead of just cleared with the background color. Looking at a glance, probably the issue is that cmd::RemoveFrame() ctor is using cmd:RemoveCel instead of cmd::ClearCel.

Reply:
Oh, yeah. I didn't see the person in this issue has background layer set up, my bad. Now I get it - I will hack a little bit and try to prepare a new solution for that.

Reply:
@dacap I came back to this yesterday, I've pushed some changes but this is still not a full solution - after I've pushed it I finally understood why there's possibility to delete frame 0 - you can delete it and basically all frames will be moved to the left. So right now my patch disabled that, so I'm working on it further.

Clearing cel on frame 0 is not a solution either. At least it's not a full solution anyways. If we clear cel on frame 0, the only thing that will change is that if we will undo action - background is not returning. The difference between ClearCel and RemoveCel is that, that RemoveCel essentially copies all insides from the cel that was before and adds it once again upon "undo" operation. It is the same cel but it is readded from internal buffer instead of reusing it from the pointer. ClearCel clears the image only, it doesn't readd the cel, it uses the same image from the previous cel. It looks like something is removing the cel.

I will let you know on days how progress with this stays, I also found some places where there are redundant if checks for example, will make them as a separate commits.